### ISSUE TITLE
Force Django to give an error on failure to render a template correctly

### ISSUE THREAD

**Issue by talumbau** [OpenSourcePolicyCenter/webapp-public#266]
@martinholmer @brendancol, I am seeing the need for a package like this:

https://github.com/counsyl/django-pedant

and not just because the README is hilarious. In the latest update of Tax-Calculator, one of the parameters changed names (from `_BE_CG_per` to `_BE_cg`). When I went to the appropriate page in a local build (in this case it was the partial equilibrium dynamic simulation input page) the field failed to render (because it was looking for the old `_BE_CG_per` which does not exist) but Django rendered the page anyway (no error generated). This is scary but is a by-product of Django's "success oriented strategy" for rendering web pages, to put it politely.

On the plus side, the test suite does fail with the new parameter, so I'm encouraged that this exact bug never would have made it out to production, but still, the lack of an error at render time is disturbing.

Do you have personal experience here and would you recommend this package or a similar tool to achieve this goal (forcing errors on bad template rendering)?

**Comment by martinholmer** [comment]
No, @talumbau, I don't have any Django or Django-Pedant experience.  But Django-Pedant sounds helpful given the way Django works.

**Comment by talumbau** [comment]
oops thanks @martinholmer I meant to tag @mistakevin.

**Comment by lucassz** [comment]
Thanks @hdoupe for pointing me to this issue. I will note that #921 has made this more necessary, as more direct calls to functions, etc. have been moved to the templates.
